src/window/editor/select_city_by_type.c

```diff
@@ -0,0 +1,183 @@
+#include "select_city_by_type.h"
+
+#include "core/lang.h"
+#include "core/string.h"
+#include "empire/city.h"
+#include "empire/type.h"
+#include "graphics/button.h"
+#include "graphics/generic_button.h"
+#include "graphics/graphics.h"
+#include "graphics/lang_text.h"
+#include "graphics/panel.h"
+#include "graphics/screen.h"
+#include "graphics/scrollbar.h"
+#include "graphics/text.h"
+#include "graphics/window.h"
+#include "input/input.h"
+#include "window/editor/map.h"
+#include "window/numeric_input.h"
+
+#include <string.h>
+
+#define MAX_BUTTONS 14
+#define BUTTON_LEFT_PADDING 32
+#define BUTTON_WIDTH 608
+#define DETAILS_Y_OFFSET 32
+#define DETAILS_ROW_HEIGHT 32
+#define MAX_VISIBLE_ROWS 14
+
+#define MAX_VALID_CITIES 500
```

Comment:
Since we can have an unlimited number of cities now, I don't think it's a good idea to hardcode a max of `500` cities.

I think it's best to use a dynamic buffer with enough space for the current number of cities, and if the new number of cities is larger you realloc the memory, but never trully delete it.

Reply:
Done.